### ISSUE TITLE
Intero + company weird behavior

### ISSUE THREAD

**Issue by monkoose** [hlissner/doom-emacs#903]
### Observed behavior
In haskell files with intero enabled, there is some strange behavior for autocompletion keymaps. When autocompletion popup is visible they are more often don't work (like `C-j` `C-n` etc), so `C-j` inserts new line etc. But sometimes it works. Don't have this behavior in any other modes/files that i have checked.

### Expected behavior
`C-n` `C-j` - company-select-next
`C-p` `C-k` - company-select-previous
as in the `bindings.el`

### Steps to reproduce
1. uncoment haskell section in your init.el
2. refresh doom and reload emacs
3. Open some haskell file try to autocomplete something with `C-j` `C-k`

### System information
emacs 26.1
develop branch of doom-emacs

**Comment by monkoose** [comment]
And seems like config below doesn't work as intended too in haskell files. It still checks file on new line creation, leaving insert mode etc.
```lisp
(after! flycheck
  (setq flycheck-check-syntax-automatically '(mode-enabled save)))
```

**Comment by hlissner** [comment]
Are you using the `+childframe` flag with the `:completion company` module?

**Comment by monkoose** [comment]
I have on `+auto` if i understand you right.

**Comment by wedens** [comment]
I experience the same issue with default config in `develop` branch (9599ca3c) with `(haskell +intero)` uncommented.

**Comment by hlissner** [comment]
@jmorag Sorry for the late followup, but the first time you use intero+haskell, if the intero package isn't installed it will do so itself, which may be the cause of that hanging.

----

I think I've zeroed in on the issue. Could any one of you give this a try:

```elisp
;; ~/.doom.d/config.el
(advice-add #'company-enable-overriding-keymap :after-while #'evil-normalize-keymaps)
```

**Comment by hlissner** [comment]
50de8a0 appears to fix the issue for me. Please update and let me know how it fares for you.

**Comment by hlissner** [comment]
I'll assume this has been resolved by 50de8a0. Let me know if that isn't the case and I will reopen the issue.
